File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/ReloadRune.java
package org.rexcellentgames.burningknight.entity.item.accessory.equippable;

public class ReloadRune extends Equippable {
	{
		sprite = "item-scroll_a";
	}

	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);
		this.owner.modifyStat("reload_time", 1f);
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);
		this.owner.modifyStat("reload_time", -1f);
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/ReloadRune.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

class ReloadRune : Equippable() {

	val time: Float
		get() = this.level.toFloat()

	init {
		sprite = "item-scroll_a"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.modifyStat("reload_time", time)
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.modifyStat("reload_time", -time)
	}

	override fun getMaxLevel(): Int {
		return 2
	}
}